### ISSUE TITLE
Address timing out (or failing) test PanzerAdaptersSTK_MixedPoissonExample-ConvTest-Hex-Order-3 test in the Trilinos-atdm-toss3-intel-debug-openmp-panzer build

### ISSUE THREAD

**Issue by bartlettroscoe** [trilinos/Trilinos#2751]
Summary: 
**CC:** @trilinos/panzer, @fryeguy52 

## Description

As can be seen in the queries

* https://testing-vm.sandia.gov/cdash/queryTests.php?project=Trilinos&filtercombine=and&filtercombine=and&filtercombine=and&filtercount=3&showfilters=1&filtercombine=and&field1=buildname&compare1=61&value1=Trilinos-atdm-toss3-intel-debug-openmp-panzer&field2=testname&compare2=65&value2=PanzerAdaptersSTK_MixedPoissonExample-ConvTest-Hex-Order-3&field3=buildstarttime&compare3=84&value3=now
* 

The test `PanzerAdaptersSTK_MixedPoissonExample-ConvTest-Hex-Order-3` is the only failing test in the panzer-only test build `Trilinos-atdm-toss3-intel-debug-openmp-panzer` over the last few days, except when there are system issues that cause many of the tests to fail (see #2699).

In all but one of the builds were this test was the only Panzer test that failed, it timed out a 10 minutes.  The one exception was on 2018-05-10 where it failed as shown at:

* https://testing-vm.sandia.gov/cdash/testDetails.php?test=46467457&build=3498830

which showed the failure.

```
--------------------------------------------------------------------------
ORTE has lost communication with its daemon located on node:

  hostname:  ser285

This is usually due to either a failure of the TCP network
connection to the node, or possibly an internal failure of
the daemon itself. We cannot recover from this failure, and
therefore will terminate the job.

--------------------------------------------------------------------------
```

## Steps to Reproduce

As described at:

* https://github.com/trilinos/Trilinos/blob/develop/cmake/std/atdm/README.md#chamaserrano

Clone the Trilinos git repo on 'serrano' and then do:

```
$ cd <some_build_dir>/

$ source $TRILINOS_DIR/cmake/std/atdm/load-env.sh intel-debug-openmp

$ cmake \
  -DTrilinos_CONFIGURE_OPTIONS_FILE:STRING=cmake/std/atdm/ATDMDevEnv.cmake \
  -DTrilinos_ENABLE_TESTS=ON -DTrilinos_ENABLE_Panzer=ON \
  $TRILINOS_DIR

$ make -j16

$ salloc -N1 --time=0:20:00 --account=<YOUR_WCID> ctest -VV -R PanzerAdaptersSTK_MixedPoissonExample-ConvTest-Hex-Order-3
```

## Related Issues

* Related to: #2699

**Comment by bartlettroscoe** [comment]
@rppawlo,

Since this test is passing just fine in the companion build `Trilinos-atdm-toss3-intel-opt-openmp-panzer` (which is about to be promoted to the "ATDM" CDash Track/Group) as shown at:

* https://testing-vm.sandia.gov/cdash/index.php?project=Trilinos&date=2018-05-15&filtercombine=and&filtercombine=and&filtercount=2&showfilters=1&filtercombine=and&field1=buildname&compare1=61&value1=Trilinos-atdm-toss3-intel-opt-openmp-panzer&field2=buildstarttime&compare2=84&value2=now

and 

* https://testing-vm.sandia.gov/cdash/queryTests.php?project=Trilinos&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercount=3&showfilters=1&filtercombine=and&field1=buildname&compare1=61&value1=Trilinos-atdm-toss3-intel-opt-openmp-panzer&field2=testname&compare2=65&value2=PanzerAdaptersSTK_MixedPoissonExample-ConvTest-Hex-Order-3&field3=buildstarttime&compare3=84&value3=now

can we just disable this test in this debug build `Trilinos-atdm-toss3-intel-debug-openmp-panzer`?

That way, we may be able to promote the `Trilinos-atdm-toss3-intel-debug-openmp-panzer` to the "ATDM" CDash Track/Group pretty soon as well.

**Comment by rppawlo** [comment]
sure. That is fine.

**Comment by jmgate** [comment]
@bartlettroscoe, can you see the recent discussion on EM-Plasma/BuildScripts!70 to see if that might resolve your issues here?

**Comment by bartlettroscoe** [comment]
> @bartlettroscoe, can you see the recent discussion on EM-Plasma/BuildScripts!70 to see if that might resolve your issues here?

@jmgate, do you mean explicitly setting `KOKKOS_ARCH=BDW`?

**Comment by jmgate** [comment]
Nope, meant the `--hint=nomultithread` stuff.  Not sure if it applies here, but it was something affecting whether or not Panzer examples were timing out.

**Comment by bartlettroscoe** [comment]
@jmgate  said:

> Nope, meant the `--hint=nomultithread` stuff. Not sure if it applies here, but it was something affecting whether or not Panzer examples were timing out.

Looks like `--hint=nomultithread` only impacts the 'mutrino' platform, not the TOSS3 'serrano' platform.  is that right?  Are you suggesting we try this on 'serrano'?

**Comment by jmgate** [comment]
No idea.  I simply saw two issues pop up with certain Panzer tests timing out&mdash;seemed like they might be related.  If not, feel free to disregard.

**Comment by bartlettroscoe** [comment]
@jmgate, the option `--hint=nomultithread` does not seem to be supported by the OpenMPI `mpiexec` program on 'serrano'.  That is, for the executable:

```
$ which mpiexec
/opt/openmpi/1.10/intel/bin/mpiexec

$ mpiexec --help | grep hint
[empty]
```

**Comment by bartlettroscoe** [comment]
After the commit 652a011 was merge on 5/16/2018, the test `PanzerAdaptersSTK_MixedPoissonExample-ConvTest-Hex-Order-3` disappeared from the build `Trilinos-atdm-toss3-intel-debug-openmp-panzer` starting 5/17/2018 as shown at:

* https://testing-vm.sandia.gov/cdash/index.php?project=Trilinos&parentid=3521979

and

* https://testing-vm.sandia.gov/cdash/viewTest.php?onlyfailed&buildid=3521980

I am marking this with "Disabled Tests" and "Stalled" so this can be fixed later, perhaps.

**Comment by fryeguy52** [comment]
This test is also timing out in the waterman builds, as show [here](https://testing-vm.sandia.gov/cdash/queryTests.php?project=Trilinos&date=2018-08-20&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercount=3&showfilters=1&filtercombine=and&field1=buildname&compare1=65&value1=Trilinos-atdm-waterman-&field2=status&compare2=62&value2=passed&field3=status&compare3=62&value3=notrun)

- Trilinos-atdm-waterman-cuda-9.2-debug
- Trilinos-atdm-waterman-cuda-9.2-opt
- Trilinos-atdm-waterman-gnu-debug-openmp

In the build `Trilinos-atdm-waterman-gnu-opt-openmp` it is only taking 42 seconds to complete

Do we want to disable it for waterman builds as well?

**Comment by bartlettroscoe** [comment]
CC: @rppawlo 

@fryeguy52 said:

> Do we want to disable it for waterman builds as well?

Yes, I think we should just disable this test in the 'debug' builds like we did on the other platforms.

**Comment by bartlettroscoe** [comment]
FYI: PR #3559 merged which disables this test on 'waterman' `cuda-9.2` builds. 

To re-enable for testing, ether configure with:

```
-DPanzerAdaptersSTK_MixedPoissonExample-ConvTest-Hex-Order-3_DISABLE=OFF
```

or create a commit that removes the `set(PanzerAdaptersSTK_MixedPoissonExample-ConvTest-Hex-Order-3_DISABLE ON)` statement that was added in commit aab6928c2c62d3be1add453ffaeff4f2f50a6f6b.

**Comment by bartlettroscoe** [comment]
On can see that on 10/9/2018 in the build `Trilinos-atdm-waterman-cuda-9.2-debug` shown [here](https://testing.sandia.gov/cdash-dev-view/viewTest.php?buildid=4029686), for example, that the test `PanzerAdaptersSTK_MixedPoissonExample-ConvTest-Hex-Order-3` is newly listed as "missing".

I will now add the "Disabled Tests" label to this issue.

**Comment by rppawlo** [comment]
you could, but I suspect you will see the same thing as the Curl Quad Order 4 - If ctest is run in parallel, then they lock up. They are certainly passing on waterman when run one at a time. I reenabled everything in my local copy.

**Comment by bartlettroscoe** [comment]
@rppawlo said:

> you could, but I suspect you will see the same thing as the Curl Quad Order 4 - If ctest is run in parallel, then they lock up. They are certainly passing on waterman when run one at a time. I reenabled everything in my local copy.

@rppawlo, is re-enabling it this just for the optimized builds on 'waterman' worth doing?  Otherwise, as shown, for example, [here](https://testing.sandia.gov/cdash/queryTests.php?project=Trilinos&date=2018-11-12&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercount=2&showfilters=1&filtercombine=and&field1=buildname&compare1=65&value1=Trilinos-atdm-&field2=testname&compare2=61&value2=PanzerAdaptersSTK_MixedPoissonExample-ConvTest-Hex-Order-3), this test is not getting run in any CUDA builds at all.

**Comment by rppawlo** [comment]
yes - just waterman would be a good test
